### PR TITLE
Add support for external metrics in kubectl

### PR DESCRIPTION
**Pull request by bskiba**
**What this PR does / why we need it**:
Add support for metrics coming from external sources when describing or printing horizontal pod autoscaler objects. API change: #60096

@MaciekPytel @DirectXMan12 

**Release note**:
```
NONE
```

### CODE REVIEW

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -2823,6 +2823,21 @@ func describeHorizontalPodAutoscaler(hpa *autoscaling.HorizontalPodAutoscaler, e
 		w.Write(LEVEL_0, "Metrics:\t( current / target )\n")
 		for i, metric := range hpa.Spec.Metrics {
 			switch metric.Type {
+			case autoscaling.ExternalMetricSourceType:
+				if metric.External.TargetAverageValue != nil {
+					current := "<unknown>"
+					if len(hpa.Status.CurrentMetrics) > i && hpa.Status.CurrentMetrics[i].External != nil {
+						current = hpa.Status.CurrentMetrics[i].External.CurrentAverageValue.String()
```
Comment (MaciekPytel) [inline pkg/printers/internalversion/describe.go]: Can you add a check for CurrentAverageValue nil here?

File: pkg/printers/internalversion/printers.go [hunk 2]
```diff
@@ -1491,6 +1491,20 @@ func formatHPAMetrics(specs []autoscaling.MetricSpec, statuses []autoscaling.Met
 	count := 0
 	for i, spec := range specs {
 		switch spec.Type {
+		case autoscaling.ExternalMetricSourceType:
+			if spec.External.TargetAverageValue != nil {
+				current := "<unknown>"
+				if len(statuses) > i && statuses[i].External != nil {
+					current = statuses[i].External.CurrentAverageValue.String()
```
Comment (MaciekPytel) [inline pkg/printers/internalversion/printers.go]: Again, check for nil CurrentAverageValue

File: pkg/printers/internalversion/describe.go (hunk 1 above)
Comment (bskiba) [reply pkg/printers/internalversion/describe.go]: Done.

File: pkg/printers/internalversion/printers.go (hunk 2 above)
Comment (bskiba) [reply pkg/printers/internalversion/printers.go]: Done.

File: pkg/printers/internalversion/describe.go [hunk 3]
```diff
@@ -2823,6 +2823,22 @@ func describeHorizontalPodAutoscaler(hpa *autoscaling.HorizontalPodAutoscaler, e
...
+						hpa.Status.CurrentMetrics[i].External.CurrentAverageValue != nil {
+						current = hpa.Status.CurrentMetrics[i].External.CurrentAverageValue.String()
+					}
+					w.Write(LEVEL_1, "%q:\t%s / %s\n", metric.External.MetricName, current, metric.External.TargetAverageValue.String())
+				} else {
+					current := "<unknown>"
+					if len(hpa.Status.CurrentMetrics) > i && hpa.Status.CurrentMetrics[i].External != nil {
+						current = hpa.Status.CurrentMetrics[i].External.CurrentValue.String()
+					}
+					w.Write(LEVEL_1, "%q:\t%s / %s\n", metric.External.MetricName, current, metric.External.TargetValue.String())
```
Comment (DirectXMan12) [inline pkg/printers/internalversion/describe.go]: need to somehow differentiate between "value" and "average value"

File: pkg/printers/internalversion/printers.go [hunk 4]
```diff
@@ -1491,6 +1491,20 @@ func formatHPAMetrics(specs []autoscaling.MetricSpec, statuses []autoscaling.Met
...
+				if len(statuses) > i && statuses[i].External != nil && statuses[i].External.CurrentAverageValue != nil {
+					current = statuses[i].External.CurrentAverageValue.String()
+				}
+				list = append(list, fmt.Sprintf("%s/%s", current, spec.External.TargetAverageValue.String()))
+			} else {
+				current := "<unknown>"
+				if len(statuses) > i && statuses[i].External != nil {
+					current = statuses[i].External.CurrentValue.String()
+				}
+				list = append(list, fmt.Sprintf("%s/%s", current, spec.External.TargetValue.String()))
```
Comment (DirectXMan12) [inline pkg/printers/internalversion/printers.go]: need to differentiate between "value" and "average value"

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
comments inline

File: pkg/printers/internalversion/describe.go (hunk 3 above)
Comment (bskiba) [reply pkg/printers/internalversion/describe.go]: Added details in brackets after metric name (following example from resource utilization).

File: pkg/printers/internalversion/printers.go (hunk 4 above)
Comment (bskiba) [reply pkg/printers/internalversion/printers.go]: Added "(avg)" to the average value. I don;t think there is room for much more (don;t want to stretch the presentation too much)

Comment (DirectXMan12) [reply pkg/printers/internalversion/printers.go]: ack, that's what I was looking for

**Review by DirectXMan12 — APPROVED** [review]

**Review by janetkuo — APPROVED** [review]

**Pull request merged**